lib/ansible/modules/cloud/azure/GUIDELINES.md

```diff
@@ -0,0 +1,155 @@
+
+# Rules for Azure Ansible facts modules
+
+## Input Parameters
+
+Every facts module should at least contain following options. In most cases these options should be sufficient:
+
+|Option|Description|Optional|
+|------|-----------|--------|
+|resource_group|Resource group name|YES|
+|name|Resource Name|YES|
+|tags|Tags that should be matched|YES|
+
+Facts module should allow users to:
+- query all the resources in a given subscription (if possible)
+- query all the resources in a specific resource group (if possible)
+- query specific resource instance
+- filter resources by tags (if possible)
+
+## Return value rules
+
+Return values should be a composite of:
+- input of the main module
+    - structure should be preserved as closely as possible
+    - fields should be omited only if impossible or difficult to retrieve
+- additional read only fields assigned by the service, for instance:
+    - resource **id** field
+    - assigned **ip_address**
+    - any other fields that may be important to the user, like URLs etc.
+
+Following rules should be followed:
+- return structure should have as short name as possible, for instance **servers**
```

Comment:
it's better to explain with sample.  "Previsouly result is structure like `ansible_fests.azure_vms`, now shorten it to `vms`. shorten naming is too general, we'd like to call out of removing `ansible_facts` in case people see old facts as samples